### ISSUE TITLE
Loose app with jsp/jsf errors based on manifest location

### ISSUE THREAD

**Issue by chyt** [OpenLiberty/open-liberty#5898]
In Open Liberty 18.0.0.3, loose applications with jsp/jsf error based on the `sourceOnDisk` location of the manifest file.

To replicate: 

1. Clone https://github.com/chyt/crossword-gen-jsf
2. Notice that `looseApplication = true` on line 75 in the `build.gradle`. 
3. Run `./gradlew build libertyRun`
4. Observe the following error in `messages.log`:

```
Caused by: java.lang.IllegalStateException: No Factories configured for this Application. This happens if the faces-initialization does not work at all - make sure that you properly include all configuration settings necessary for a basic faces application and that all the necessary libs are included. Also check the logging output of your web application and your container for any exceptions!
If you did that and find nothing, the mistake might be due to the fact that you use some special web-containers which do not support registering context-listeners via TLD files and a context listener is not setup in your web.xml.
```

When hitting the endpoint, you will see:

```
Error 500: javax.servlet.ServletException: SRVE0207E: Uncaught initialization exception created by servlet
```

Change `looseApplication = false` and re-build, and observe that the application starts successfully with neither of the two error messages being displayed. 

This is the content of the loose app xml:

```
<?xml version="1.0" encoding="UTF-8"?>
<archive>
    <dir sourceOnDisk="/Users/ctianus.ibm.com/Desktop/crossword-gen-jsf/src/main/webapp" targetInArchive="/"/>
    <dir sourceOnDisk="/Users/ctianus.ibm.com/Desktop/crossword-gen-jsf/build/classes/java/main" targetInArchive="/WEB-INF/classes/"/>
    <dir sourceOnDisk="/Users/ctianus.ibm.com/Desktop/crossword-gen-jsf/build/resources/main" targetInArchive="/WEB-INF/classes/"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/27.0-jre/c6ad87d2575af8ac8ec38e28e75aefa882cc3a1f/guava-27.0-jre.jar" targetInArchive="/WEB-INF/lib/guava-27.0-jre.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-math3/3.6.1/e4ba98f1d4b3c80ec46392f25e094a6a2e58fcbf/commons-math3-3.6.1.jar" targetInArchive="/WEB-INF/lib/commons-math3-3.6.1.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/org.primefaces/primefaces/6.2/784f963a01788b4c57cd12b5bb21529adf287649/primefaces-6.2.jar" targetInArchive="/WEB-INF/lib/primefaces-6.2.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/com.google.guava/failureaccess/1.0/50bacb0bdbbeeda25142aff57e2214c402e1193c/failureaccess-1.0.jar" targetInArchive="/WEB-INF/lib/failureaccess-1.0.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/b421526c5f297295adef1c886e5246c39d4ac629/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar" targetInArchive="/WEB-INF/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/com.google.code.findbugs/jsr305/3.0.2/25ea2e8b0c338a877313bd4672d3fe056ea78f0d/jsr305-3.0.2.jar" targetInArchive="/WEB-INF/lib/jsr305-3.0.2.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/org.checkerframework/checker-qual/2.5.2/cea74543d5904a30861a61b4643a5f2bb372efc4/checker-qual-2.5.2.jar" targetInArchive="/WEB-INF/lib/checker-qual-2.5.2.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/com.google.errorprone/error_prone_annotations/2.2.0/88e3c593e9b3586e1c6177f89267da6fc6986f0c/error_prone_annotations-2.2.0.jar" targetInArchive="/WEB-INF/lib/error_prone_annotations-2.2.0.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/com.google.j2objc/j2objc-annotations/1.1/976d8d30bebc251db406f2bdb3eb01962b5685b3/j2objc-annotations-1.1.jar" targetInArchive="/WEB-INF/lib/j2objc-annotations-1.1.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/.gradle/caches/modules-2/files-2.1/org.codehaus.mojo/animal-sniffer-annotations/1.17/f97ce6decaea32b36101e37979f8b647f00681fb/animal-sniffer-annotations-1.17.jar" targetInArchive="/WEB-INF/lib/animal-sniffer-annotations-1.17.jar"/>
    <file sourceOnDisk="/Users/ctianus.ibm.com/Desktop/crossword-gen-jsf/build/tmp/war/MANIFEST.MF" targetInArchive="/META-INF/MANIFEST.MF"/>
</archive>
```

Attached is a set of logs with `com.ibm.ws.artifact*=FINEST` tracing.
[logs.zip](https://github.com/OpenLiberty/open-liberty/files/2634358/logs.zip)

**Comment by chyt** [comment]
This problem can be avoided using one of the following workarounds, both involving the `sourceOnDisk` and `targetInArchive` values of the manifest file:

1. Move the manifest `sourceOnDisk` inside a `META-INF` folder. 
2. Remove the leading slash from the `targetInArchive` value (`/META-INF/MANIFEST.MF` to `META-INF/MANIFEST.MF`)

It is unclear to me why either of these workarounds would resolve the issue, while the combination of mappings for the manifest attached above would cause failure. For some reason, the internal implementation seems to care about the parent directory of the manifest file in the source. Also not sure if it is appropriate to use a relative path in the `targetInArchive` value, as this does not seem to be the convention.

**Comment by volosied** [comment]
This has been fixed in 2.2 and 2.3  in OL.
